ui/components/multichain/edit-accounts-modal/edit-accounts-modal.tsx

```diff
@@ -181,7 +188,29 @@ export const EditAccountsModal: React.FC<EditAccountsModalProps> = ({
                     <ButtonPrimary
                       data-testid="connect-more-accounts-button"
                       onClick={() => {
+                        const addedAccounts = selectedAccountAddresses.filter(
```

Comment:
this should be a faster and more efficient approach:
```
// Convert the arrays to Sets for O(1) lookup
const defaultSet = new Set(defaultSelectedAccountAddresses);
const selectedSet = new Set(selectedAccountAddresses);

// Get accounts that are in `selectedAccountAddresses` but not in `defaultSelectedAccountAddresses`
const addedAccounts = selectedAccountAddresses.filter(
  (address) => !defaultSet.has(address),
);

// Get accounts that are in `defaultSelectedAccountAddresses` but not in `selectedAccountAddresses`
const removedAccounts = defaultSelectedAccountAddresses.filter(
  (address) => !selectedSet.has(address),
);
```

Reply:
Updated here https://github.com/MetaMask/metamask-extension/pull/27820/commits/69389a8f9e8d17afd7c48c5c3f3b10ca7847e070